sklearn/semi_supervised/_self_training.py

```diff
@@ -293,105 +318,181 @@ def fit(self, X, y):
             self.termination_condition_ = "all_labeled"
 
         self.base_estimator_.fit(
-            X[safe_mask(X, has_label)], self.transduction_[has_label]
+            X[safe_mask(X, has_label)],
+            self.transduction_[has_label],
+            **routed_params.base_estimator.fit,
         )
         self.classes_ = self.base_estimator_.classes_
         return self
 
     @available_if(_estimator_has("predict"))
-    def predict(self, X):
+    def predict(self, X, **params):
         """Predict the classes of `X`.
 
         Parameters
         ----------
         X : {array-like, sparse matrix} of shape (n_samples, n_features)
             Array representing the data.
 
+        **params : dict of str -> object
+            Parameters to pass to the underlying estimator's predict method.
+
+            .. versionadded:: 1.5
+
+            See :ref:`Metadata Routing User Guide <metadata_routing>` for more
+            details.
```

Comment:
no new line, and the comment should be the same as `fit`

Reply:
Done.